frame/election-provider-support/src/lib.rs

```diff
@@ -218,6 +218,14 @@ pub trait ElectionDataProvider {
 	///
 	/// This should be implemented as a self-weighing function. The implementor should register its
 	/// appropriate weight at the end of execution with the system pallet directly.
+	///
+	/// We don't want to have a scenario where the return of `desired_targets`, call it `y`,
+	/// is bigger than the size of the `Vec` returned by `targets`, call it `x`.
+	/// If this were to happen, the call-site should emit a warning and use `min(x, y)` as
+	/// opposed to `y`. Please check `create_snapshot_external` in `election-provider-multi-phase`
+	/// for more details.
+	///
+	/// This is documented further in issue: <https://github.com/paritytech/substrate/issues/9478>
```

Comment:
Sorry, I am a bit strict on documentation, and want to add a few suggestions here: 

1. Preferably don't reference `election-provider-multi-phase` here. The existence of this pallet should not be assumed here. 
2. Avoid first person plural, as it is not commonly used in the docs.
3. Describe what has to be done, or what is expected, not what our intentions or rationale are. A trait implementer need to know (foremost) what they must do.  

```suggestion
	/// A sensible implementation should use the minimum between this value and [`Self::targets().len()`], since desiring a winner set larger than candidates is not feasible.  
	///
	/// This is documented further in issue: <https://github.com/paritytech/substrate/issues/9478>
```